src/main/java/com/coursemngsys/coursemanagementsystem/hibernateControllers/CourseHibernateController.java

```diff
@@ -0,0 +1,116 @@
+package com.coursemngsys.coursemanagementsystem.hibernateControllers;
+
+import com.coursemngsys.coursemanagementsystem.Model.Course;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+import javax.persistence.Query;
+import javax.persistence.criteria.CriteriaQuery;
+import java.util.List;
+
+public class CourseHibernateController {
+    private SessionFactory factory = null;

```

Comment:
Nereikalingas "null" reikšmės priskyrimas, kadangi klasė turi vienintelį konstruktorių ir jame šiam kintamajam reikšmė suteikiama. Pašalinti reikšmės priskyrimą.
```suggestion
    private SessionFactory factory;

```